### PR TITLE
[WC-2391] Add custom content events prop

### PR DESCRIPTION
**Pull request by iobuhov**
### Pull request type

New feature (non-breaking change which adds functionality)

---

### Description
Add prop to control event propagation on custom content.

### CODE REVIEW

File: packages/pluggableWidgets/datagrid-web/src/Datagrid.xml [hunk 1]
```diff
@@ -136,6 +136,10 @@
                                     <enumerationValue key="no">No</enumerationValue>
                                 </enumerationValues>
                             </property>
+                            <property key="allowEventPropagation" type="boolean" defaultValue="true">
+                                <caption>Allow default events</caption>
+                                <description>If set to yes, then all default evens, such as "on click" or selection, will be triggered when the user interacts with custom content.</description>
+                            </property>
```
Comment (r0b1n) [inline packages/pluggableWidgets/datagrid-web/src/Datagrid.xml:142]: I think we never use word _events_, should call this actions probably. Something like "Allow row actions", or maybe better invert the meaning like "Block row actions".

Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/Datagrid.xml:142]: I also was thinking firs of actions, but then I changed my mind as there also selection and IMO it's better to use more generic "events".

Comment (r0b1n) [reply packages/pluggableWidgets/datagrid-web/src/Datagrid.xml:142]: Yeah, true. Let's align with @takumagott on the names/descriptions.

File: packages/pluggableWidgets/datagrid-web/src/Datagrid.editorConfig.ts
[the remark this answers is not in the document]
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/Datagrid.editorConfig.ts:1]: Yeah, good one, I forgot about it.

File: packages/pluggableWidgets/datagrid-web/src/helpers/state/column/ColumnStore.tsx [hunk 2]
```diff
@@ -262,10 +266,21 @@ const onKeyDown = (event: React.KeyboardEvent): void => {
     event.stopPropagation();
 };
 
-function CustomContent(props: { children: ReactNode }): ReactElement {
+function CustomContent({
```
[the remark this answers is not in the document]
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/helpers/state/column/ColumnStore.tsx:269]: Done

File: packages/pluggableWidgets/datagrid-web/src/Datagrid.xml (hunk 1 above)
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-web/src/Datagrid.xml]: done

Comment (gjulivan) [reply packages/pluggableWidgets/datagrid-web/src/Datagrid.xml]: check again on the suggested caption.
it should be `<caption>Allow row events</caption>`

**Review by gjulivan — DISMISSED** [review]

**Review by leonardomendix — DISMISSED** [review]

**Review by leonardomendix — APPROVED** [review]

**Review by gjulivan — APPROVED** [review]

**Pull request merged**